chimneyCats/src/main/scala/io/scalaland/chimney/cats/package.scala

```diff
@@ -15,10 +17,34 @@ trait CatsImplicits extends LowPriorityImplicits {
 
   implicit def TransformerFValidatedNelSupport[E]: TransformerFSupport[ValidatedNel[E, +*]] =
     TransformerFValidatedSupport[NonEmptyList[E]](implicitly)
+
+  implicit def TransformerFIorNecSupport[E]: TransformerFSupport[IorNec[E, +*]] =
+    TransformerFIorSupport[NonEmptyChain[E]](implicitly)
+
+  implicit def TransformerFIorNelSupport[E]: TransformerFSupport[IorNel[E, +*]] =
+    TransformerFIorSupport[NonEmptyList[E]](implicitly)
+
+  implicit def TransformerFIorNesSupport[E]: TransformerFSupport[IorNes[E, +*]] =
+    TransformerFIorSupport[NonEmptySet[E]](implicitly)
 }
 
 trait LowPriorityImplicits {
 
+  implicit def TransformerFIorSupport[EE: Semigroup]: TransformerFSupport[Ior[EE, +*]] =
+    new TransformerFSupport[Ior[EE, +*]] {
+      override def pure[A](value: A): Ior[EE, A] =
+        Ior.right(value)
+
+      override def product[A, B](fa: Ior[EE, A], fb: => Ior[EE, B]): Ior[EE, (A, B)] =
+        fa.flatMap(a => fb.map(b => (a, b)))
+
+      override def map[A, B](fa: Ior[EE, A], f: A => B): Ior[EE, B] =
+        fa.map(f)
+
+      override def traverse[M, A, B](it: Iterator[A], f: A => Ior[EE, B])(implicit fac: Factory[B, M]): Ior[EE, M] =
+        Traverse[List].traverse[Ior[EE, *], A, B](it.toList)(f).map(fac.fromSpecific(_))
```

Comment:
There might be some laziness concerns in converting the iterator to a list. Maybe we can go through a lazy list?

If that works, then we can probably generalize the entire instance to Applicative, similarly to what I wanted to do in #170

Reply:
Thanks to some direction from @kubukoz, we can do this lazily using

```scala
override def traverse[M, A, B](it: Iterator[A], f: A => Ior[EE, B])(implicit fac: Factory[B, M]): Ior[EE, M] =
  it.foldRight(Eval.always(pure(CompatLazyList.empty[B]))) { (next: A, acc: Eval[Ior[EE, CompatLazyList[B]]]) =>
      Apply[Ior[EE, *]].map2Eval(f(next), acc)(_ #:: _)
  }.value.map(fac.fromSpecific(_))
```

Reply:
Is `it.foldRight` intentional in your implementation? In my understanding it destroys any laziness concerns.

Reply:
So I've mimicked the [Cats implementation]((https://github.com/typelevel/cats/blob/master/core/src/main/scala-2.13%2B/cats/instances/lazyList.scala#L61)) for traverse on lazy list. 
As far as I understand, Cats is using Eval to be lazy but I could be wrong

Reply:
Initially, I used a tail-recursive function (for early termination) with a Vector accumulator (as I needed to append) and terminated as soon as I had a Ior.Left and then at the very, I can switched it to the builder. Would you like me to switch back to this?

Reply:
@krzemin 
```scala
      override def traverse[M, A, B](it: Iterator[A], f: A => Ior[EE, B])(implicit fac: Factory[B, M]): Ior[EE, M] = {
        @tailrec
        def go(rem: Iterator[A], acc: Ior[EE, Vector[B]]): Ior[EE, M] =
          rem.nextOption() match {
            case Some(a) =>
              val newAcc = Apply[Ior[EE, *]].map2(acc, f(a))(_ :+ _)
              go(rem, newAcc)

            case None =>
              println(acc)
              acc.map(fac.fromSpecific(_))
          }
        go(it, Ior.right(Vector.empty))
      }
```

Reply:
That mimicks the same semantics as foldRight + Eval, let me know if you want me to swap it out for this

Reply:
Ok, let's clarify few things:
- laziness in case of accessing provided `Iterator` is not very much a concern, as in most of the cases it contains data already present in memory (case classes fields)
- in fact the problem was that intermediate data structure (a `List`) was created from the iterator, leading to unnecessary garbage
- your tailrec implementation is very much similar to `foldLeft`, but the intermediate `Vector` is used for buffering (appending) results, that leads to unnecessary garbage as well; `LazyList` based implementation has the same issue 

Intermediate garbage can be avoided by:
1. consuming the iterator elements directly
2. using scala collections builder directly

Examples of such `traverse` implementations are available [for `Either`](https://github.com/scalalandio/chimney/blob/master/chimney/src/main/scala/io/scalaland/chimney/TransformerFSupport.scala#L134) and [for `Option`](https://github.com/scalalandio/chimney/blob/master/chimney/src/main/scala/io/scalaland/chimney/TransformerFSupport.scala#L88).

Reply:
Ah I understand what you are saying now @krzemin and I realized the tail-recursive implementation didn't terminate early, Here's the improved version:
```scala
      override def traverse[M, A, B](it: Iterator[A], f: A => Ior[EE, B])(implicit fac: Factory[B, M]): Ior[EE, M] = {
        @tailrec
        def go(rem: Iterator[A], acc: Ior[EE, Vector[B]]): Ior[EE, M] =
          rem.nextOption() match {
            case Some(a) =>
              val newAcc = Apply[Ior[EE, *]].map2(acc, f(a))(_ :+ _)
              
              if (newAcc.isLeft) newAcc.map(fac.fromSpecific(_))
              else go(rem, newAcc)

            case None =>
              acc.map(fac.fromSpecific(_))
          }
        go(it, Ior.right(Vector.empty))
      }
```

Let me take a look at the traverse for Either and Option that you provided and see if I can get rid of the intermediate Vector

Reply:
Hello @krzemin, sorry to keep bothering you, I took a look at the implementations for Either and Option. I noticed in the `Either` implementation, you have a type constraint on the error result `C[X] <: IterableOnce[X]`. However, I don't see a way to get rid of the intermediate `Vector` in the general case for `Ior` specifically since this allows us to support any `EE` that has a `Semigroup`instance as opposed to limiting the left side to `IterableOnce`.

Reply:
Why can't you just have `var ee: EE` and accumulate on it with Semigroup? In the while loop, depending on what you get, you can:
- Left -> terminate early with the EE value got here
- Both -> accumulate warning on `ee` and add the element to the builder
- Right -> just add the element to the builder

After the loop, if you didn't terminate early, you need to check if you need to return Both or Right (in both cases you call `.result()` on builder to obtain final collection).

I totally agree that we shouldn't limit to `IterableOnce` as it's not the cats' way.

Reply:
Oops @krzemin turns out that I have forgotten how to write mutable code :laughing:, this works:
```scala
      override def traverse[M, A, B](it: Iterator[A], f: A => Ior[EE, B])(implicit fac: Factory[B, M]): Ior[EE, M] = {
        var leftSide: EE = null.asInstanceOf[EE]
        val rightSide = fac.newBuilder
        var earlyTermination = false

        while (it.hasNext && !earlyTermination) {
          f(it.next()) match {
            case Ior.Left(a) =>
              if (leftSide == null) leftSide = a
              else leftSide = Semigroup[EE].combine(leftSide, a)
              earlyTermination = true

            case Ior.Right(b) =>
              rightSide += b

            case Ior.Both(a, b) =>
              if (leftSide == null) leftSide = a
              else leftSide = Semigroup[EE].combine(leftSide, a)
              rightSide += b
          }
        }

        if (earlyTermination) Ior.left(leftSide)
        else if (leftSide != null) Ior.both(leftSide, rightSide.result())
        else Ior.right(rightSide.result())
      }
```

Reply:
Hahaha, exactly what I meant 😄

Reply:
hehe sweet! thank you so much @krzemin 
I'll incorporate this in